rules/framework.bzl

```diff
@@ -724,7 +725,7 @@ def _bundle_dynamic_framework(ctx, is_extension_safe, avoid_deps):
             label_name = label.name,
             platform_prerequisites = platform_prerequisites,
             predeclared_outputs = predeclared_outputs,
-            product_type = rule_descriptor.product_type,
+            product_type = product_type if product_type else ctx.attr._product_type,
```

Comment:
@jerrymarino let me know if my interpretation here is correct and we don't want static as a type here. I don't know all the details about `link_dynamic` don't wanted to make sure this change is consistent with the original intent of this flag.

Reply:
It's probably fine to just hardcode this to the dynamic product type on this line specifically because it's only for `_bundle_dynamic framework`(s).

That said, we might need to contend with setting set `_product_type` argument too

Reply:
It's probably fine to just hardcode this to the dynamic product type on this line specifically because it's only for `_bundle_dynamic framework`(s).

That said, we might need to contend with setting set `_product_type` attr too?

Reply:
We can remove the default value from the private _product_type and the set it explicitly in the rule?

Reply:
https://github.com/bazel-ios/rules_ios/pull/741/commits/c209ea96a3ccd838c37a94cfc42a77345bb7a20f